test/unit/org/apache/cassandra/net/ConnectionTest.java

```diff
@@ -192,30 +195,54 @@ Settings override(Settings settings)
 
     // 30 is used for CNDB compatibility
     static final AcceptVersions legacy = new AcceptVersions(VERSION_3014, VERSION_3014);
+    static final AcceptVersions ds10 = new AcceptVersions(minimum_version, VERSION_DS_10);
+    static final AcceptVersions ds11 = new AcceptVersions(minimum_version, VERSION_DS_11);
+    static final AcceptVersions current = new AcceptVersions(current_version, current_version);
 
-    static final List<Function<Settings, Settings>> MODIFIERS = ImmutableList.of(
+    static final List<Function<Settings, Settings>> MESSAGGING_VERSIONS = ImmutableList.of(
         settings -> settings.outbound(outbound -> outbound.withAcceptVersions(legacy))
                             .inbound(inbound -> inbound.withAcceptMessaging(legacy)),
+        // Mismatched versions (in both directions) to ensure both peers will still agree on the same version.
+        settings -> settings.outbound(outbound -> outbound.withAcceptVersions(ds11))
+                            .inbound(inbound -> inbound.withAcceptMessaging(ds10)),
+        settings -> settings.outbound(outbound -> outbound.withAcceptVersions(ds10))
+                            .inbound(inbound -> inbound.withAcceptMessaging(ds11)),
+        // This setting ensures that we cover the current case for the power set where no versions are overridden.
+        settings -> settings.outbound(outbound -> outbound.withAcceptVersions(current))
+                            .inbound(inbound -> inbound.withAcceptMessaging(current))
+    );
+
+
+    static final List<Function<Settings, Settings>> MODIFIERS = ImmutableList.of(
         settings -> settings.outbound(outbound -> outbound.withEncryption(encryptionOptions))
                             .inbound(inbound -> inbound.withEncryption(encryptionOptions)),
         settings -> settings.outbound(outbound -> outbound.withFraming(LZ4))
     );
 
+    // Messaging versions are a kind of modifier, but they can only be applied once per setting, so they are broken
+    // out into a separate list.
     static final List<Settings> SETTINGS = applyPowerSet(
```

Comment:
The extra complexity of these settings appears to be an issue for some tests. In looking closer, I can see that the rpc timeout is affecting some, like `testMessagePurging`. I think it might also affect `testCloseIfEndpointDown`. For example, calling `DatabaseDescriptor.setRpcTimeout(5000L);`, speeds up these tests a bit. The test failures appear to be timeouts, so maybe there is something to look into here. I haven't seen legitimate failures yet.

Reply:
For reference, the delay in `testMessagePurging` comes from this block:

https://github.com/datastax/cassandra/blob/19bd463e78a8f064e9227a949ea7d61f61cb98c4/src/java/org/apache/cassandra/net/OutboundConnection.java#L1089-L1109